### ISSUE TITLE
Return only the top level record on data creation

### ISSUE THREAD

**Issue by kiaking** [vuex-orm/vuex-orm#286]
Currently, creation methods such as `insert` and `update` return all entities which were created. For example;

```js
// This is going to return...
User.create({
  data: {
    id: 1,
    posts: [
      { id: 1 }
    ]
  }
})

// This.
{
  users: [{ id: 1 }],
  posts: [{ id: 1 }]
}
```

The reason why it returns like this, instead of just returning a user record, is because when you have nested child relationship, for example, the user has many posts that belong to the author (user), then it was hard to determine which record was actually the top level entity.

But obviously, this is cumbersome to use. Cause you have to do this to actually access the data.

```
const result = User.create({ ... })

result.users[0]
```

I found that the `normalizr` obtains the top level record as its result. We can use that determine only the top level record and can return only them.

So the new return value would be like this.

```
User.create({
  data: {
    id: 1,
    posts: [
      { id: 1 }
    ]
  }
})

User {
  id: 1
}
```

There were several requests that this is how return value should look like. However, since this is breaking change to the API so I would like to have some opinion from the community.

What do you guys think?

**Comment by bertBruynooghe** [comment]
I do not think this is a real problem, as destructuring solves this problem rather nicely, I think.

Your first code snippet can be written as:

```
const { users: [user], posts: [post] } = await User.create({
  data: {
    id: 1,
    posts: [
      { id: 1 }
    ]
  }
})
```

And from now on you can refer to `user` and `post` without bothering about the complexity of the object at creation. 

I now have a number of situations where I'm rather interested in one of the children, so I wouldn't welcome the change as a good one...

Also, your second snippet could be rewritten as:

```
const { users: [user] } = User.create({ ... })
```

which I don't find that hard to read.

A simplification that improves readability I could live with is by using an array of arrays, which would read as:

```
const [[user], posts] =  await User.create({
  data: {
    id: 1,
    posts: [
      { id: 1 }
    ]
  }
})
```

and
```
const [[user]] = User.create( {... })
```

but then you might probably run into questions on the order of types in the array, or I am wrong?

**Comment by kiaking** [comment]
Thank you so much for the feedback! I wasn't thinking about destructing. Yeah, your example looks really neet and I kinda dig it.

I see your point as well on how you're interested in child relations. However, there's a thing I'm concerning about is that the entities you get as a return value from `create` or `insert` method merges all nested same entity.

I mean if you insert "Users" that have "Posts" that have "Users", then you get the last "Users" as well. In other words, there's no way to determine which record was the top level records.

I too think to get all entities are useful. So I'm currently thinking to add that method as well, where one method will return only the top level entity, and another method returns whole entities (same as current behavior). Haven't come up with method name yet though...
